### ISSUE TITLE
MEGAsync - Windows Explorer Context Menu Missing, Status Icons Missing - Update Required

### ISSUE THREAD

**Issue by Psyyche** [meganz/MEGAsync#995]
## Issue Description / Details
I have a paid plan (**Pro Lite**) which I make very frequent use of. I am using **Windows 10 Professional**. Having tested on various computers I can verify these issues are not restricted to any one machine. I have already created a support ticket (**#RSY-860-96352**) and was assured that the issues would be passed along to developers, however I also wanted to include my findings where others can review them openly. I am adding this here to provide the required details for anyone else who may be facing these particular issues until they are resolved. Know you are **not** insane. Things are clearly not working as expected and others are indeed facing the same problems.

After MEGA is installed you are supposed to see new context menu options added to your interface including **"Upload to MEGA"**, **"Get MEGA Link"**, **"View on MEGA"** etc. included in the **context menu** to allow for convenient access to your stored files from within the main file explorer window - these do not appear in the context menu within the file explorer where they should anymore. Reinstalling MEGA and/or toggling the desired options within the settings of the program itself does not seem to fix this behavior..

Additionally, the **Windows Explorer sync status icons** that are meant to indicate the status of a particular file or folder within your storage do not seem to be functional either. Screenshots included for further reference..

Official documentation about sync icons..
![Status Icons - Official](https://github.com/user-attachments/assets/c94fbd53-78c9-454a-aaed-8339dc033775)

Visuals of expected behavior..
![Status Icons, Links](https://github.com/user-attachments/assets/19e955a2-8145-4b69-80c0-c98d65c7b201)

Visual of MEGA program settings..
![Settings](https://github.com/user-attachments/assets/351435ed-8ddf-4d70-b970-ed9b7fd4d283)

Version/about information for reference..
![5 4 1](https://github.com/user-attachments/assets/90321e6a-fa00-4107-87c8-f3e9d2e25da6)

## Steps to Reproduce
> 1.) Download the latest version of the MEGA installer (https://mega.io/desktop#download)..
> 2.) Install the program and set up your preferred sync.
> 3.) Navigate to MEGA's directory on your local drive.
> 4.) Right click any item within MEGA's directory. Observe there are no options added by MEGA as anticipated.
> 5.) Observe there are no icons showing on files or folders to indicate their sync status in MEGA.

## Actual Behavior
Mega sync icons and context menu entries are missing from the file explorer - Windows Explorer context menus and sync status icons are not present. Reinstalling MEGA as administrator or otherwise does not serve to fix these issues.

## Expected Behavior
Mega sync status icons and context menu entries are supposed to be present in Windows Explorer..

## Additional Notes For Development
As for the issue with programs registering overlay icons in the system - that is a Windows design choice and is usually the root cause of that particular issue. It can be overcome by updating the code of MEGA to instead resort to using more modern methods such as the API features provided by Microsoft. Various other sync services are already taking advantage of this API to help eliminate issues regarding overlay icons (sync.com - just to name one popular option as a reference to test against). As you can see by visiting the links below this method allows Windows Explorer to display status information as it's own 'status' category, instead of directly overlaying the icons. As a result programs can serve up the same information while avoiding the overlay icon restrictions built into the system. I recommend that MEGA consider using this API to their advantage in the future..

The new “Cloud files API” in Windows 10 version 1709 provides this alternative: _“In addition to the standard Windows state icons used for hydration state, you can provide custom state icons for additional service-specific properties. Replaces legacy icon overlay Shell extensions.”_

https://superuser.com/questions/1166577/why-does-windows-limit-icon-overlays-to-15
https://learn.microsoft.com/en-us/windows/win32/cfapi/build-a-cloud-file-sync-engine

**Comment by vtmateos** [comment]
Thanks for the detailed report, it is really appreciated. We are planning to integrate Cloud files API capabilities within MEGAsync in mid term. Please stay tuned for future updates! 
Thanks.

**Comment by Psyyche** [comment]
Yes sir. Thank you very much for the quick response here. MEGAsync remains one of the best cloud sync options available to date and I am always glad to provide support & constructive thoughts. In the interim I can make due using my online panel directly for most features. I will patiently await future updates. Blessings!

**-P5YCH3**

**Comment by jwallaceqlp** [comment]
This might seem like a very silly suggestion - but when you installed it as admin, did you install it for ALL USERS on the computer?  A good way to check is open Settings -> Apps, and see if Mega was listed. If it is not, it probably wasn't installed for all users. 

If "only install for this user" was selected, the registry changes won't be made for the other users on the computer. 

I hope that makes sense, this happened to me in the wild, and took me awhile to figure out.

**Comment by Psyyche** [comment]
Yes, that makes good sense although it does not apply here. The computers I installed with only have a single user account (therefore even in this scenario that one account would still be receiving said registry changes). I can clearly see the registry changes are made on the computer by viewing them manually because I know where the registry values for MEGA overlay icons are stored, alongside the others in the system..

**Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\ShellIconOverlayIdentifiers\\**
![Registry](https://github.com/user-attachments/assets/ea269e22-f3de-4a16-b8db-6f26a1fdbc1e)

You can see the registry values are present after installation. I have personally attempted installation both ways as suggested as well to ensure my own sanity and due diligence.

![1](https://github.com/user-attachments/assets/3a6ca14f-d88a-4dd4-85d4-2ddf55e2893a)

Either way, the issues in question should not be issues effecting the end users. The developers have reassured me that updates are on the way to repair intended functions, including the system overlay icons and the context menu entries. Features promoted should work regardless of user preferences selected during setup. I will patiently await future updates. Blessings!

**-P5YCH3**

**Comment by Psyyche** [comment]
Yes, that makes good sense.. I will reinstall once again using that option and post results. The computers I installed with prior only have a single user account (therefore even in this scenario that one account should still be receiving said registry changes). I can clearly see the registry changes are made on the computer by viewing them manually because I know where the registry values for MEGA overlay icons are stored, alongside the others in the system..

**Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\ShellIconOverlayIdentifiers\\**

![Registry](https://github.com/user-attachments/assets/ea269e22-f3de-4a16-b8db-6f26a1fdbc1e)

You can see the registry values are present after installation. You will also observe that when running the latest installer fresh, the default selection appears to be _'Install just for me'_..

![2](https://github.com/user-attachments/assets/c419bb62-f159-418a-a707-16337f00f46f)

I find this strange if the end result is that the user will potentially have broken elements to their installation because of permissions etc.. Either way, the issues in question should not be issues effecting the end users. Features installed should work regardless of user preferences selected during setup.. I have personally attempted installation both ways as suggested as well to ensure my own sanity and due diligence.

![1](https://github.com/user-attachments/assets/3a6ca14f-d88a-4dd4-85d4-2ddf55e2893a)

Ignoring the default selection and choosing 'Install for anyone using this computer'.. It would appear that reinstalling using this option has fixed the system overlay icons as well as the context menu entires. Thank you [jwallaceqlp](https://github.com/jwallaceqlp) for the excellent suggestion. I am actually slightly taken aback that this is the case! This option should obviously come with a warning during installation as it causes issues with these features. Everything is now functioning on a test computer as expected. Blessings.

![3](https://github.com/user-attachments/assets/a066cf64-1d00-4ddb-b16b-10716e67e44e)
![4](https://github.com/user-attachments/assets/883c6e39-688a-4126-8098-48699970391e)

**-P5YCH3**

**Comment by Psyyche** [comment]
This issue marked closed as the mystery has been thoroughly solved, to much elation. Have fun with your data everyone! :) 👍

**Comment by ChaseCarlson** [comment]
Hi all,

I encountered this issue with the default selection which I kept, and was left confused until I found this thread (lucky me!) After reinstalling for all users, I was able to get the sync icons.

However, for user experience, would it not make more sense to default to the other selection (install for all users), and provide a warning about this behavior during install?

Another option would be what other applications like Dropbox do, which circumvents a registry edit entirely (using Cloud Filter API cfapi.h, available on Windows 10 version 1709 or later); use the "Status" field on Windows to display the sync status as such:

<img width="223" height="244" alt="Image" src="https://github.com/user-attachments/assets/75fae13d-516b-4137-b995-969f59f2cb66" />

Regardless, I'm grateful to have found a solution. I would consider adding a change to the installer flow to at least inform the user about this issue.

Thanks
Chase

**Comment by Psyyche** [comment]
**+1** to that excellent notion. I posted the same sentiments in September of 2024, a little over a year ago. We need this subtle but important change to be implemented. It would certainly eliminate a lot of undo confusion and testing for everyone who might run into the problem, and it is not hard to modify the default value of that option and add some text as a warning to the next rendition of the installer. This should have been implemented already if you want the truth - so I will Reopen the issue in the hopes that perhaps someone on the development team might take notice - not only of this as a pertinent issue to address but also so that perhaps we can receive an update on said implementation of the Cloud files API capabilities.. 

I would still absolutely love the ability to select which behavior is actively running (Windows Cloud Files API or System Overlay Icons). An option to toggle between which behavior is used to determine the display within Windows Explorer in the **Settings** (and the Windows Cloud files API as a suggested default during initial **setup**) would go a long way to eliminate any potential future issues. 

[Psyyche](https://github.com/Psyyche)
> As for the issue with programs registering overlay icons in the system - that is a Windows design choice and is usually the root cause of that particular issue. It can be overcome by updating the code of MEGA to instead resort to using more modern methods such as the API features provided by Microsoft. Various other sync services are already taking advantage of this API to help eliminate issues regarding overlay icons (sync.com - just to name one popular option as a reference to test against). As you can see by visiting the links below this method allows Windows Explorer to display status information as it's own 'status' category, instead of directly overlaying the icons. As a result programs can serve up the same information while avoiding the overlay icon restrictions built into the system. I recommend that MEGA consider using this API to their advantage in the future..
> 
> The new “Cloud files API” in Windows 10 version 1709 provides this alternative: _“In addition to the standard Windows state icons used for hydration state, you can provide custom state icons for additional service-specific properties. Replaces legacy icon overlay Shell extensions.”_
> 
> https://superuser.com/questions/1166577/why-does-windows-limit-icon-overlays-to-15 https://learn.microsoft.com/en-us/windows/win32/cfapi/build-a-cloud-file-sync-engine

[vtmateos](https://github.com/vtmateos)
> Thanks for the detailed report, it is really appreciated. We are planning to integrate Cloud files API capabilities within MEGAsync in mid term. Please stay tuned for future updates! Thanks.

**-P5YCH3**
